### PR TITLE
Source Google Sheets: exposes row batch size config

### PR DESCRIPTION
**Pull request by akulgoel96**
## What
Currently, the `ROW_BATCH_SIZE` value is a constant defined in the code and in our case it was leading to a lot of rate limit exceeded errors while reading spreadsheets having 100,000+ rows.

## How
This solution exposes the above variable to the connector, while keeping the default value as `200` itself to maintain backwards compatibility.

## 🚨 User Impact 🚨
As mentioned above, it is backwards compatible since the default value has been kept the same. So, there should not be any user impact.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@misteryeo can you check this change in the specification of Source Google Sheets?

File: airbyte-integrations/connectors/source-google-sheets/google_sheets_source/google_sheets_source.py [hunk 1]
```diff
@@ -132,6 +130,7 @@ def read(
         sheet_to_column_name = Helpers.parse_sheet_and_column_names_from_catalog(catalog)
         spreadsheet_id = Helpers.get_spreadsheet_id(config["spreadsheet_id"])
 
+        row_batch_size = config['row_batch_size']
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-sheets/google_sheets_source/google_sheets_source.py:133]:
```suggestion
        row_batch_size = config.get('row_batch_size', ROW_BATCH_SIZE)
```
Let the default batch in the code to not create backward compability and also let the default in the spec.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
There isn;t any docs in Google saying about a better row size?

File: airbyte-integrations/connectors/source-google-sheets/google_sheets_source/spec.yaml [hunk 2]
```diff
@@ -15,6 +15,11 @@ connectionSpecification:
         Enter the link to the Google spreadsheet you want to sync
       examples:
         - https://docs.google.com/spreadsheets/d/1hLd9Qqti3UyLXZB2aFfUWDT7BG-arw2xy4HR3D-dwUb/edit
+    row_batch_size:
+      type: integer
+      title: Row Batch Size
+      description: Number of rows fetched when making a google sheet API call
```
Comment (misteryeo) [inline airbyte-integrations/connectors/source-google-sheets/google_sheets_source/spec.yaml:21]:
```suggestion
      description: Number of rows fetched when making a Google Sheet API call. Defaults to 200.
```

**Review by misteryeo — CHANGES_REQUESTED** [review]

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**